packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -609,23 +602,24 @@ function initializeMode(vm: VM) {
         );
     }
 
+    // Default modes.
+    vm.renderMode = RenderMode.Shadow;
+    vm.shadowMode = ShadowMode.Native;
+
     if (renderMode === 'light') {
         vm.renderMode = RenderMode.Light;
-        vm.shadowMode = ShadowMode.None;
-        return;
+    } else if (renderer.ssr) {
+        vm.renderMode = RenderMode.Shadow;
     }
 
     if (renderer.ssr) {
-        vm.renderMode = RenderMode.Shadow;
         vm.shadowMode = ShadowMode.Native;
-        return;
+    } else if (renderer.syntheticShadow) {
```

Comment:
We are checking twice `renderer.ssr` in the middle of the hot path to reset the default value. This looks wasteful to me.

Reply:
This is the equivalent with a single `renderer.ssr` check:

```
    // Default modes.
    vm.renderMode = RenderMode.Shadow;
    vm.shadowMode = ShadowMode.Native;

    if (renderer.ssr) {
        vm.renderMode = RenderMode.Shadow;
        vm.shadowMode = ShadowMode.Native;
    } else if (renderer.syntheticShadow) {
        vm.shadowMode =
            isTrue(preferNativeShadow) && isNativeShadowRootDefined
                ? ShadowMode.Native
                : ShadowMode.Synthetic;
    }

    if (renderMode === 'light') {
        vm.renderMode = RenderMode.Light;
    }
```
It seems less readable since it's less clear that `renderMode === 'light'` has higher priority than `renderer.ssr`, but I can make this change if you think it's worth it.

Reply:
I'm fine either way. Another option is to cache the result of the `===` check in a const, and use the cost in two places.

Reply:
Thanks @pmdartus, changed it to:
```
    if (renderMode === 'light') {
        vm.renderMode = RenderMode.Light;
    } else {
        vm.renderMode = RenderMode.Shadow;
    }

    if (renderer.ssr) {
        vm.shadowMode = ShadowMode.Native;
    } else if (renderer.syntheticShadow) {
        vm.shadowMode =
            isTrue(preferNativeShadow) && isNativeShadowRootDefined
                ? ShadowMode.Native
                : ShadowMode.Synthetic;
    } else {
        vm.shadowMode = ShadowMode.Native;
    }
```